rmw_connextdds_common/src/ndds/dds_api_ndds.cpp

```diff
@@ -197,25 +200,119 @@ rmw_connextdds_initialize_participant_qos_impl(
   dp_qos->user_object.topic_user_object.size = sizeof(void *);
   dp_qos->user_object.content_filtered_topic_user_object.size = sizeof(void *);
 #endif /* RMW_CONNEXT_SHARE_DDS_ENTITIES_WITH_CPP */
+  if (!ctx->domain_tag) {
+    const auto pid = rcutils_get_pid();
+    ctx->domain_tag = DDS_String_alloc(18 + log10(pid));
+    sprintf(ctx->domain_tag, "ros_discovery_off_%d", pid);
```

Comment:
This would feel much safer to me if we just created the string we want, then take the size of it to allocate into the domain tag, because if someone updated the string `"ros_discovery..."` whatever to something else, and forgot to update the alloc line above... that would be bad and it would be easy to do.

Reply:
I agree, the string should be declared as a static constant and the length should not be a "magic number".
Also, the return value for `DDS_String_alloc()` must be checked.

```suggestion
    static const std::string domain_tag_pfx = "ros_discovery_off_";
    const auto pid = rcutils_get_pid();
    const auto domain_tag_len = domain_tag_pfx.length() + log10(pid);
    ctx->domain_tag = DDS_String_alloc(domain_tag_len);
    if (nullptr == ctx->domain_tag) {
      RMW_CONNEXT_LOG_ERROR_SET("failed to allocate domain tag string");
      return RMW_RET_ERROR;
    }
    sprintf(ctx->domain_tag, "%s%d", domain_tag_pfx.c_str(), pid);
```